File: member-service/src/main/java/com/hedvig/memberservice/commands/EditMemberInformationCommand.java
package com.hedvig.memberservice.commands;

import com.hedvig.memberservice.web.dto.InternalMember;
import lombok.Value;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

@Value
public class EditMemberInformationCommand {

  @TargetAggregateIdentifier Long id;

  InternalMember member;
  String token;

  public EditMemberInformationCommand(String id, InternalMember updatedMember, String token) {
    this.id = Long.parseLong(id);
    this.member = updatedMember;
    this.token = token;
  }
}

File: member-service/src/main/java/com/hedvig/memberservice/commands/EditMemberInformationCommand.kt
package com.hedvig.memberservice.commands

import com.hedvig.memberservice.web.dto.InternalMember
import org.axonframework.commandhandling.TargetAggregateIdentifier

data class EditMemberInformationCommand(
    @TargetAggregateIdentifier
    val id: String,
    val member: InternalMember,
    val token: String
)
